File: Client/tecmo-tourney/src/app/core/models/gameResult.model.ts
import { GameStatus, GameType } from 'src/app/enums';
import {IGameResultPlayer} from './gameResultPlayer.model';

export interface IGameResult {
    gameResultId: number;    
    tournamentId: number;
    player1: IGameResultPlayer;
    player2: IGameResultPlayer;
    date: Date;
    status: GameStatus;
    gameType: GameType;
    /** If set, this game does not count toward this player's preliminary seeding. */
    seedingExemptPlayerId?: number | null;
    /** UTC when the match was marked in progress at the game station. */
    gameStartedAt?: string | null;
  }

File: Service/Models/GameResultModel.cs
namespace TecmoTourney.Models
{
    public class GameResultModel
    {
        public int GameResultId { get; set; }
        public GameResultStatsModel Player1 { get; set; } = new GameResultStatsModel();
        public GameResultStatsModel Player2 { get; set; } = new GameResultStatsModel();
        public int TournamentId { get; set; }
        public GameStatus Status { get; set; }
        public GameType GameType { get; set; }
        public int BracketGameId { get; set; }
        public int MatchUpIndex { get; set; }
        public DateTime Date { get; set; }
        /// <summary>If set, this game does not count toward this player's preliminary seeding.</summary>
        public int? SeedingExemptPlayerId { get; set; }
    }
}
